src/Rings/MPolyQuo.jl

```diff
@@ -648,16 +654,17 @@ lift(a::MPolyQuoElem) = a.f
 (Q::MPolyQuo)() = MPolyQuoElem(Q.R(), Q)
 
 function (Q::MPolyQuo)(a::MPolyQuoElem)
-   @assert parent(a) == Q
+   parent(a) !== Q && error("Parent mismatch")
    return a
 end
 
-function (Q::MPolyQuo)(a::MPolyElem) 
-   if base_ring(Q) === parent(a)
-      return MPolyQuoElem(a, Q)
-   else
-      return MPolyQuoElem(Q.R(a), Q)
-   end
+function (Q::MPolyQuo{S})(a::S) where {S <: MPolyElem}
+   base_ring(Q) === parent(a) || error("Parent mismatch")
+   return MPolyQuoElem(a, Q)
+end
```

Comment:
The juliaification of this `if else end` introduces a functional change. Now if the `a` has the right type (`S`) but the wrong parent, then we can't get coersion. This is my only gripe, and if this is intended, then everything is fine here.

Reply:
This is intended.

P.S.: I am actually dreaming of some `try_coerce(R, x)::Tuple{Bool, RingElement}` methods which whould get rid of many of these dirty hacks and at the same time allow useful error messages.